modules/openapi-generator/src/main/resources/dart2/dart_constructor.mustache

```diff
@@ -2,9 +2,9 @@
   {{{classname}}}({
   {{#vars}}
     {{!
-        A field is @required in Dart when it is
-        required && !nullable && !defaultValue in OAS
+        A field is required in Dart when it is
+        required && !defaultValue in OAS
```

Comment:
This is the correct evaluation, and it tallies with other templates (like Kotlin).

Clarification: although a property is `required`, it doesn't mean that it cannot receive a `null` value. The need to use `required` in this case is purely to attract the developer to send a value (knowingly) even if it's `null` – it just means that the developer is aware and has chosen to send a `null`.